rules/Symfony73/Rector/Class_/AddVoteArgumentToVoteOnAttributeRector.php

```diff
@@ -0,0 +1,120 @@
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+                        use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+                        use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+
+                        final class MyVoter extends Voter
+                        {
+                            protected function supports(string $attribute, mixed $subject): bool
+                            {
+                                return true;
+                            }
+
+                            protected function voteOnAttribute(string $attribute, $subject, TokenInterface $token): bool
+                            {
+                                return true;
+                            }
+                        }
+                        CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+                        use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+                        use Symfony\Component\Security\Core\Authorization\Voter\Voter;
+
+                        final class MyVoter extends Voter
+                        {
+                            protected function supports(string $attribute, mixed $subject): bool
+                            {
+                                return true;
+                            }
+
+                            protected function voteOnAttribute(string $attribute, $subject, TokenInterface $token, ?\Symfony\Component\Security\Core\Authorization\Voter\Vote $vote = null): bool
+                            {
+                                return true;
+                            }
+                        }
+                        CODE_SAMPLE
+                    ,
+                ),
+            ],
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $classMethod = null;
+
+        if ($node->extends !== null && $this->isName($node->extends, self::VOTER_CLASS)) {
+            $classMethod = $node->getMethod('voteOnAttribute');
+        }
+
+        if ($classMethod === null) {
+            foreach ($node->implements as $implement) {
+                if ($this->isName($implement, self::VOTE_INTERFACE)) {
+                    $classMethod = $node->getMethod('vote');
+                    break;
+                }
+            }
+        }
+
+        if ($classMethod === null) {
+            return null;
+        }
+
+        if (count($classMethod->params) !== 3) {
+            return null;
+        }
+
+        $classMethod->params[] = new Param(
+            new Variable('vote'),
+            new ConstFetch(new Name('null')),
+            new Node\NullableType(new FullyQualified(self::VOTE_CLASS))
+        );
+
+        return null;
```

Comment:
use return $node on change:

```suggestion
        return $node;
```

Reply:
fixed.